### PR TITLE
Bug fix in pyromixin for posterior sampling

### PR DESCRIPTION
**Pull request by vitkl**
The method `._get_obs_plate_sites()` should discard observed variables (unless requested otherwise) and plates. Current code always performs posterior sampling even if unobserved local variables do not exist in the model, e.i. code never enters this statement:
```
if len(obs_plate_sites) == 0:
    # if no local variables - don't sample
    break
```

This PR addresses the issue by adding a check for observed and plates.

### CODE REVIEW

**Comment by vitkl** [comment]
@adamgayoso please have a look at this

File: scvi/model/base/_pyromixin.py [hunk 1]
```diff
@@ -262,7 +262,7 @@ def _get_obs_plate_return_sites(self, return_sites, obs_plate_sites):
         else:
             return obs_plate_sites
 
-    def _get_obs_plate_sites(self, args, kwargs):
+    def _get_obs_plate_sites(self, args, kwargs, sample_observed):
```
Comment (adamgayoso) [inline scvi/model/base/_pyromixin.py:265]: can you document this function and add relevant typing?

**Review by adamgayoso — CHANGES_REQUESTED** [review]
LGTM outside of the small request for docstring

**Comment by vitkl** [comment]
I think this is done now.

File: scvi/model/base/_pyromixin.py [hunk 2]
```diff
@@ -262,7 +270,12 @@ def _get_obs_plate_return_sites(self, return_sites, obs_plate_sites):
         else:
             return obs_plate_sites
 
-    def _get_obs_plate_sites(self, args, kwargs):
+    def _get_obs_plate_sites(
+        self,
+        args: Optional[list],
+        kwargs: Optional[dict],
```
Comment (adamgayoso) [inline scvi/model/base/_pyromixin.py:276]: usually optional has a None default. Don't believe this is the case here?

**Review by adamgayoso — CHANGES_REQUESTED** [review]

**Comment by vitkl** [comment]
Yes, you are right actually - these arguments are not optional and don't
have default values. I will change this.

File: scvi/model/base/_pyromixin.py (hunk 2 above)
Comment (adamgayoso) [reply scvi/model/base/_pyromixin.py:276]:
```suggestion
        args: list,
        kwargs: dict,
```

**Review by adamgayoso — APPROVED** [review]

**Pull request merged**